core/store/orm/orm.go

```diff
@@ -1488,6 +1488,53 @@ func toISO8601(t time.Time) string {
 		t.Year(), t.Month(), t.Day(), t.Hour(), t.Minute(), t.Second(), t.Nanosecond(), tz)
 }
 
+const removeUnstartedTransactionsQuery = `
+WITH
+unstarted_job_runs AS (
+	DELETE FROM job_runs
+	WHERE status = 'unstarted'
+	RETURNING id, run_request_id
+),
+linked_run_requests AS (
+	DELETE FROM run_requests AS rr
+	WHERE rr.id IN (
+		SELECT run_request_id
+		FROM unstarted_job_runs
+	)
+),
+linked_task_runs AS (
+	DELETE FROM task_runs AS ts
+	WHERE ts.job_run_id IN (
+		SELECT id
+		FROM unstarted_job_runs
+	)
+	RETURNING id
+),
+linked_eth_txes AS (
+	SELECT et.id
+	FROM eth_txes AS et, eth_task_run_txes AS etrt, linked_task_runs AS ltr
+	WHERE et.id = etrt.eth_tx_id
+	AND etrt.task_run_id = ltr.id
+),
+removed_eth_txes AS (
+	DELETE FROM eth_txes AS et
+	WHERE et.id IN (
+		SELECT id
+		FROM linked_eth_txes
+	)
+	RETURNING id
+)
+DELETE FROM eth_tx_attempts AS eta
```

Comment:
I'm 99% sure this is covered by the foreign key cascade. Have you checked foreign keys to reduce the size of this statement?

Reply:
You're correct! I removed the section of query that removes eth_tx_attempts and they are still getting removed by deletes in eth_txes. 👍